subprojects/docs/src/samples/userguide/tutorial/manifest/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,53 @@
+plugins {
+    java
+}
+version = "1.0"
+
+// tag::add-to-manifest[]
+tasks.getByName<Jar>("jar") {
+    manifest {
+        attributes(
+            "Implementation-Title" to "Gradle",
+            "Implementation-Version" to version
+        )
+    }
+}
+// end::add-to-manifest[]
+
+// tag::custom-manifest[]
+val sharedManifest = the<JavaPluginConvention>().manifest {
+    attributes (
+        "Implementation-Title" to "Gradle",
+        "Implementation-Version" to version
+    )
+}
+
+task<Jar>("fooJar") {
+    manifest = project.the<JavaPluginConvention>().manifest {
+        from(sharedManifest)
+    }
+}
+// end::custom-manifest[]
+
+// tag::merge[]
+task<Jar>("barJar") {
+    manifest {
+        attributes("key1" to "value1")
+        from(sharedManifest, "src/config/basemanifest.txt")
+        from(listOf("src/config/javabasemanifest.txt", "src/config/libbasemanifest.txt"), closureOf<ManifestMergeSpec> {
```

Comment:
:x: now that #6574 is fixed, could you merge `master` onto your branch and use the `Action<T>` taking method here?

Reply:
Done.